client/src/components/MapInfoMenu/MapInfoMenu.js

```diff
@@ -72,8 +115,8 @@ export const MapInfoMenu = ({
           </Typography>
         </CardContent>
         <CardActions disableSpacing>
-          <IconButton aria-label="add to favorites">
-            <FavoriteIcon />
+          <IconButton aria-label="add to favorites" onClick={onFavouriteToggle}>
+            <FavoriteIcon sx={{ color: isFavourite ? "red" : "grey" }} />
```

Comment:
move to separate const` const FavoriteIconColor = isFavourite ? "red" : "grey"`;